push-build.sh

```diff
@@ -96,7 +97,19 @@ RELEASE_BUCKET=${FLAGS_bucket:-"kubernetes-release-dev"}
 [[ $KUBE_GCS_UPDATE_LATEST == "n" ]] && FLAGS_noupdatelatest=1
 
 KUBECTL_OUTPUT=$(cluster/kubectl.sh version --client 2>&1 || true)
-if [[ "$KUBECTL_OUTPUT" =~ GitVersion:\"(${VER_REGEX[release]}(\.${VER_REGEX[build]})?)\", ]]; then
+if [[ "$KUBECTL_OUTPUT" =~ GitVersion:\"(${VER_REGEX[release]}(\.${VER_REGEX[build]})?-dirty)\", ]]; then
+  if ((FLAGS_allow_dirty)); then
+    logecho "The current build is marked dirty, but pushing anyway because --allow-dirty"
+    logecho "was given. This option should not be used for releases."
+    LATEST=${BASH_REMATCH[1]}
+  else
+    logecho "The current build is marked dirty. If this is a test build, you can"
+    logecho "specify --allow-dirty to enable pushing it."
+    logecho
+    logecho "Please note that --allow-dirty should never be used for releases."
```

Comment:
yeah, my bad, I mis-remembered where push-build.sh was used. `anago` uses the same `release::gcs` functions, but does its own checks and calls them directly.